delete previous image on cover change

Ahad-patel opened the pull request:
<!---
Thank you for submitting a pull request to AppFlowy. The team will dedicate their best efforts to reviewing and approving your pull request. If you have any questions about the project or feedback for us, please join our [Discord](https://discord.gg/wdjWUXXhtw).
-->

<!---
If your pull request adds a new feature, please drag and drop a video into this section to showcase what you've done! If not, you may delete this section.
-->

### Feature Preview

<!---
List at least one issue here that this PR addresses. If it fixes the issue, please use the [fixes](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) keyword to close the issue. For example:
fixes https://github.com/AppFlowy-IO/AppFlowy/pull/2106
-->

<!---
Before you mark this PR ready for review, run through this checklist!
-->

Closes: https://github.com/AppFlowy-IO/AppFlowy/issues/5145
 
I have created two methods for deletePreviousImageFromCloudStorage and deleteFile in DocumentSevce Class
when ever the user changes cover image the this method will check if there the previous image was file, If so then it will delete the previous image. 

I have a question ther is a an method DocumentEventDeleteFile which takes DownloadFilePB payload which has two params localFilePath and url, the url can be obtained from coverDetails but localFilepath is not stored any where how do i delete it from the local storage?  @Xazin 

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AppFlowy-IO/AppFlowy?pullRequest=6368) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AppFlowy-IO/AppFlowy?pullRequest=6368) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AppFlowy-IO/AppFlowy?pullRequest=6368) it.</sub>

Ahad-patel commented:
Hey @LucasXu0, Is my approach to this issue is correct?

Ahad-patel commented on frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/image/image_util.dart:
```diff
@@ -117,3 +117,7 @@ Future<List<ImageBlockData>> extractAndUploadImages(
 }
+
+Future<void> deleteImageFromLocalStorage(String localImagePath) async {
+  await File(localImagePath).delete();
```
done

Ahad-patel commented:
Hey @Xazin, I was trying to add test case in this file => https://github.com/Ahad-patel/AppFlowy/blob/delete-unecessary-images/frontend/appflowy_flutter/integration_test/desktop/document/document_with_cover_image_test.dart#L8-L9, but stuck on how should I approach the local file upload case, Because the file will be present in my machine but might not be availaible in some one else machine, Is there any similar code shoud I look into?

Ahad-patel commented:
hey @LucasXu0 , @Xazin any updates here?

LucasXu0 commented:
@Ahad-patel In my opinion, the deletion seems unable to be mocked in the test. A workaounrd is to add the flag for the test to detect when the delete function is called while changing the cover image. If the function is called, we can assume the deletion takes effect.

Ahad-patel commented:
@LucasXu0 , I have added test case for local image upload and added a variable `deleteImageTestCounter` to test if `deleteImageFromLocalStorage` is being called when we remove/change cover image.
